File: python3/0001 two-sum.py
"""
Last updated: 9/18/2025, 1:54:35 PM
URL: https://leetcode.com/problems/two-sum/
Runtime: 3 ms (Beats 50.08%)
Memory: 19.2 MB (Beats 10.32%)
Language: python3
"""

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        pair_idx = {}

        for i, num in enumerate(nums):
            if target - num in pair_idx:
             return [i, pair_idx[target - num]]
            pair_idx[num] = i
        


        

File: java/0001 two-sum.java
/**
 * Last updated: 9/18/2025, 8:38:32 PM
 * URL: https://leetcode.com/problems/two-sum/
 * Runtime: 2 ms (Beats 98.89%)
 * Memory: 45.1 MB (Beats 31.42%)
 * Language: java
 */

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pairIdx = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (pairIdx.containsKey(target - num)) {
                return new int[] { i, pairIdx.get(target - num) };
            }
            pairIdx.put(num, i);
        }

        return new int[] {};        
    }
}